ext/include/opentelemetry/ext/http/common/url_parser.h

```diff
@@ -118,21 +119,21 @@ class UrlParser
       pos = url_.find('?', cpos);
       if (pos == std::string::npos)
       {
-        path_  = std::string(url_.begin() + cpos, url_.begin() + url_.length());
+        path_  = url_.substr(cpos, url_.length());
```

Comment:
Here and below: strictly speaking, it should be `(cpos, url_.length - cpos)` but because we want all remaining characters, we can save a few CPU cycles by not subtracting `cpos`.

Reply:
Even simpler, when we want all the remaining substring:

```
path_ = url_.substr(cpos);
```

https://en.cppreference.com/w/cpp/string/basic_string/substr

> (e.g. if count == npos), the returned substring is [pos, [size()](https://en.cppreference.com/w/cpp/string/basic_string/size))

Reply:
Updated the code.

Reply:
Thanks, much cleaner. Waiting for CI to complete, then merging.